### PR TITLE
Added documentation for pull request

### PR DESCRIPTION
**Pull request by ojorgensen**

### CODE REVIEW

File: ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py [hunk 1]
```diff
@@ -1,97 +1,86 @@
...
         self._cols = cols
         self._num_agents = num_agents
 
-    def return_blank_board(self) -> Array:
+    def return_blank_board(self) -> chex.Array:
         return jnp.zeros((self._rows, self._cols), dtype=int)
 
-    def pick_start(self, key: PRNGKey, layout: Array, wire_id: int, max_length: int) -> Wire:
+    def pick_start(
+            self, key: chex.PRNGKey, grid: chex.Array, wire_id: int, max_length: int
```
Comment (mwolinska) [inline ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py:33]: incorrect indentation, double check section 3.4 here : https://google.github.io/styleguide/pyguide.html#s3.4-indentation

File: ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py [hunk 2]
```diff
@@ -253,85 +280,89 @@ def add_agents(self, key: PRNGKey, board: Array, max_length: int):
...
         
         Returns:
-            board: jnp.Array with wires added (or a blank board if generation failed)
+            board: jnp.chex.Array with wires added (or a blank board if generation failed).
         """
-        def body_fun(self, max_length_int, i, state):
-            """
-            Main loop. Checks whether a board has been successfully generated, generates a new board if it hasn't.
+        def try_to_generate(
+                self, max_length_int: int, i: int, state: Tuple[chex.Array, bool]
```
Comment (mwolinska) [inline ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py:297]: incorrect indentation

**Review by mwolinska — COMMENTED** [review]
Thanks for your work! Please address comments and then good to merge. You can also run the black package on the file

File: ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py [hunk 3]
```diff
@@ -1,97 +1,86 @@
...
-#from ic_routing_board_generation.board_generator.abstract_board import AbstractBoard
-import numpy as np
-from copy import deepcopy
-# import random
 from typing import List, Tuple
 import jax.numpy as jnp
 import jax
+import chex
 
 from ic_routing_board_generation.board_generator.jax_board_generation.grid_jax import Grid
```
[the remark this answers is not in the document]
Comment (ojorgensen) [reply ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py:10]: I import this for typing purposes (they use chex for PRNGKeys in other parts of their codebase), so think I'll probably leave it in for this

File: ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py [hunk 4]
```diff
@@ -1,97 +1,86 @@
...
     stack_push
 
-"""
-#from jax.numpy import asarray # Currently jaxlib is not supported on windows.  This will have to be sorted.
-#from env_viewer import RoutingViewer # Currently jaxlib is not supported on windows.  This will have to be sorted.
-# import jumanji.environments.combinatorial.routing.constants
-#from jumanji.environments.combinatorial.routing.constants import TARGET, HEAD, EMPTY
-#from jumanji.environments.combinatorial.routing.constants import SOURCE as WIRE
-"""
 EMPTY, PATH, POSITION, TARGET = 0, 1, 2, 3
```
[the remark this answers is not in the document]
Comment (ojorgensen) [reply ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py:14]: Done

File: ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py [hunk 5]
```diff
@@ -1,97 +1,86 @@
...
-        return (key, layout, new_wire, wire_id, max_length), can_start
+        new_wire, grid = jax.lax.cond(can_start, inner_create_wire, lambda _: (dummy_wire, grid), key)
+        return (key, grid, new_wire, wire_id, max_length), can_start
 
-    def adjacent_cells(self, cell: int) -> Array:
-        """
-        Given a cell, return a jnp.Array of size 4 with the flat indices of
-        adjacent cells. Padded with -1's if less than 4 adjacent cells (if on the edge of the grid)
+    def adjacent_cells(self, cell: int) -> chex.Array:
+        """Given a cell, return a jnp.chex.Array of size 4 with the flat indices of
```
[the remark this answers is not in the document]
Comment (ojorgensen) [reply ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py:75]: Done!

File: ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py [hunk 6]
```diff
@@ -1,97 +1,86 @@
...
-#from jumanji.environments.combinatorial.routing.constants import SOURCE as WIRE
-"""
 EMPTY, PATH, POSITION, TARGET = 0, 1, 2, 3
-#  HEAD, TARGET, WIRE, EMPTY = 4,3,2,0
-STARTING_POSITION = POSITION  # My internal variable to disambiguate the word "position"
-# Also available to import from constants NOOP, LEFT, LEFT, UP, RIGHT, DOWN
-
-
 @dataclass
 class Position:
```
[the remark this answers is not in the document]
Comment (ojorgensen) [reply ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py:16]: Done!

File: ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py (hunk 2 above)
Comment (ojorgensen) [reply ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py:297]: Done!

File: ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py (hunk 1 above)
Comment (ojorgensen) [reply ic_routing_board_generation/board_generator/jax_board_generation/most_basic_rw.py:33]: Done!

**Pull request merged**